internal/pkg/manifest/env.go

```diff
@@ -69,6 +71,73 @@ type environmentVPCConfig struct {
 	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
 }
 
+type environmentCDNConfig struct {
+	EnableCDN *bool
+	CDNConfig AdvancedCDNConfig // mutually exclusive with EnableCDN
+}
+
+// AdvancedCDNConfig represents an advanced configuration for a Content Delivery Network.
+type AdvancedCDNConfig struct {
+	PrefixListIngress *bool `yaml:"public_ingress_allowed"`
```

Comment:
I wonder if this field will be clear enough to customers, what do you think of flipping the statement so that the field is by default false?
```suggestion
	LimitIngressToCDN *bool `yaml:"limit_origin_ingress"` 
```

Reply:
Will keep in mind as part of why I'm now splitting this PR is so that I can move this field to the environment ALB (in environmentHTTPConfig) so as to more accurately reflect where the security group resides. And yes, personally I think it's better to have public ingress disabled by default, so I'll change this for the next PR after splitting this one.